src/main/ruby/truffleruby/core/type.rb

```diff
@@ -145,8 +145,12 @@ def self.rb_num2dbl(val)
         val.to_f
       elsif Primitive.is_a?(val, Rational)
         val.to_f
+      elsif val === true
+        raise TypeError, 'no implicit conversion to float from true'
+      elsif val === false
+        raise TypeError, 'no implicit conversion to float from false'
       elsif Primitive.is_a?(val, String)
-        raise TypeError, 'no implicit conversion from to float from string'
+        raise TypeError, 'no implicit conversion to float from string'
```

Comment:
Could we have new tests covering these changes (in `rb_num2dbl`)?

Reply:
Looking into it.